src/noInapplicableLifeCycleRule.ts

```diff
@@ -0,0 +1,109 @@
+import * as Lint from 'tslint';
+import * as ts from 'typescript';
+import * as SyntaxKind from './util/syntaxKind';
+import {sprintf} from 'sprintf-js';
+import {Ng2Walker} from './angular/ng2Walker';
+import {difference, intersection} from './util/setOperations';
+
+
+function getInterfaceNames(t: any): string {
+  if (t.expression && t.expression.name) {
+    return t.expression.name.text;
+  }
+  return t.expression.text;
+}
+
+function getInterfaces(node: ts.ClassDeclaration): any[] {
```

Comment:
The behavior could be incorrect in case this class is base for a component, i.e.:

```
class A implements OnInit {...}

@Component(...)
class B extends A {...}
```